### PR TITLE
[Ready] Hey why dont I port that one donut pr? Ye just the one food related pr... Just the one... Oh no

### LINKED ISSUE #9952: Port all the new donuts from /tg/
image

### PR DESCRIPTION
**Pull request by Trilbyspaceclone**
8 hours later.
## About The Pull Request

ports a lot things!
Grills, Snowcone fixes, more American Empire food, BBQ stuff, trash, icons, and more
tgstation/tgstation#44419
tgstation/tgstation#46220
tgstation/tgstation#44914
tgstation/tgstation#46933
tgstation/tgstation#48153
tgstation/tgstation#46898
tgstation/tgstation#46924
tgstation/tgstation#48050
close #9952 Over did this Request I think
## Why It's Good For The Game

Food stuffs and staying more closer to TG.

## Changelog
:cl:
add: ports all the new donuts, burgars, and chicken stuff from RG
add: ports new snowcone
add: ports grill
add: ports beakfeast tag/mood lit as TG has it
add: ports all the amazing new sprites
tweak: ports crafting for many things like snowcones needing water
balance: ports of many craftings
fix: fixed a few things
soundadd: lowers fryers sound
imageadd: ported icons for new food/grill
imagedel: ports the deletion of some icons and images
spellcheck: ports a spell check for the snowcones
code: ports fixes for stuff I didnt know were even broken with snowcones
/:cl:

### CODE REVIEW

**Comment by YakumoChen** [comment]
I warned you not to port the food with the plants we don't have. Did you listen? No.

**Comment by GrayRachnid** [comment]
https://i.imgur.com/Uyx68IY.png

i wouldn't ignore travis

**Comment by Trilbyspaceclone** [comment]
@GrayRachnid That has nothing to do with the PR/porting ive done.
Its to do with something different that Ghommie made/ported and its asking forth IDs to hold paths rather then Paths for IDs.
It just so happen to pull a bad luck for the week and will Rng fail people

File: code/modules/cargo/packs/organic.dm [hunk 1]
```diff
@@ -178,6 +178,23 @@
...
+	contains = list(/obj/item/stack/sheet/mineral/coal/five,
+					/obj/machinery/grill/unwrenched)
+	crate_name = "grilling starter kit crate"
+
+/datum/supply_pack/organic/grillfuel
+	name = "Grilling Fuel Kit"
+	desc = "Contains coal and coal accessories. (Note: only ten coal sheets.)"
+	cost = 1000
+	crate_type = /obj/structure/closet/crate
+	contains = list(/obj/item/stack/sheet/mineral/coal/ten,)
```
Comment (Ghommie) [inline code/modules/cargo/packs/organic.dm]:
```suggestion
	contains = list(/obj/item/stack/sheet/mineral/coal/ten)
```

File: code/modules/food_and_drinks/food/snacks_pastry.dm [hunk 2]
```diff
@@ -35,72 +38,291 @@
 	if(last_check_time + 50 < world.time)
 		if(ishuman(M))
 			var/mob/living/carbon/human/H = M
-			if(M.mind && HAS_TRAIT(M.mind, TRAIT_LAW_ENFORCEMENT_METABOLISM) && !HAS_TRAIT(H, TRAIT_AGEUSIA))
+			if(HAS_TRAIT(H.mind, TRAIT_LAW_ENFORCEMENT_METABOLISM) && !HAS_TRAIT(H, TRAIT_AGEUSIA))
```
Comment (Ghommie) [inline code/modules/food_and_drinks/food/snacks_pastry.dm]: Keep the mind existence check, it helps against runtimes should it be fed to a mindless mob.

File: code/modules/food_and_drinks/food/snacks_pastry.dm [hunk 3]
```diff
@@ -35,72 +38,291 @@
...
-	extra_reagent = "cherryjelly"
-	foodtype = JUNKFOOD | GRAIN | FRIED | FRUIT
-
-/obj/item/reagent_containers/food/snacks/donut/meat
-	bonus_reagents = list("ketchup" = 1)
-	list_reagents = list("nutriment" = 3, "ketchup" = 2)
-	tastes = list("meat" = 1)
-	foodtype = JUNKFOOD | MEAT | GROSS | FRIED
-
-/obj/item/reagent_containers/food/snacks/donut/semen
```
Comment (Ghommie) [inline code/modules/food_and_drinks/food/snacks_pastry.dm:92]: :+1:
If someone ever wants to readd them, I guess they'll also have to bring their sprites up to date now.

File: code/modules/food_and_drinks/food/snacks_soup.dm [hunk 4]
```diff
@@ -225,3 +225,22 @@
...
+	list_reagents = list("nutriment" = 3, "liquidelectricity" = 5)
+	tastes = list("mushroom" = 1, "electrons" = 4, "shockingly good")
+	filling_color = "#CC2B52"
+	foodtype = VEGETABLES | TOXIC
+
+/obj/item/reagent_containers/food/snacks/soup/bungocurry
+	name = "bungo curry"
+	desc = "A spicy vegetable curry made with the humble bungo fruit, Exotic!"
+	icon_state = "bungocurry"
+	bonus_reagents = list("vitamin" = 5, "bungojuice" = 6)
```
Comment (Ghommie) [inline code/modules/food_and_drinks/food/snacks_soup.dm]: You forgot to add bungo juice from the same PR that updates donuts.

File: code/modules/food_and_drinks/kitchen_machinery/grill.dm [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+				grilled_item = I
+				to_chat(user, "<span class='notice'>You put the [grilled_item] on [src].</span>")
+				update_icon()
+				grill_loop.start()
+				return
+		else
+			if(I.reagents.has_reagent(/datum/reagent/consumable/monkey_energy))
+				grill_fuel += (20 * (I.reagents.get_reagent_amount(/datum/reagent/consumable/monkey_energy)))
+				to_chat(user, "<span class='notice'>You pour the Monkey Energy in [src].</span>")
+				I.reagents.remove_reagent(/datum/reagent/consumable/monkey_energy, I.reagents.get_reagent_amount(/datum/reagent/consumable/monkey_energy))
```
Comment (Ghommie) [inline code/modules/food_and_drinks/kitchen_machinery/grill.dm]:
```suggestion
				I.reagents.remove_reagent("monkey_energy", I.reagents.get_reagent_amount("monkey_energy"))
```
IDs to typepath haven't been PR'd yet.

File: code/modules/food_and_drinks/kitchen_machinery/grill.dm [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+				return
+			else if(!grilled_item && user.transferItemToLoc(I, src))
+				grilled_item = I
+				to_chat(user, "<span class='notice'>You put the [grilled_item] on [src].</span>")
+				update_icon()
+				grill_loop.start()
+				return
+		else
+			if(I.reagents.has_reagent(/datum/reagent/consumable/monkey_energy))
+				grill_fuel += (20 * (I.reagents.get_reagent_amount(/datum/reagent/consumable/monkey_energy)))
```
Comment (Ghommie) [inline code/modules/food_and_drinks/kitchen_machinery/grill.dm]:
```suggestion
				grill_fuel += (20 * (I.reagents.get_reagent_amount("monkey_energy")))
```

File: code/modules/food_and_drinks/kitchen_machinery/grill.dm [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+		return
+	else
+		grill_fuel -= 1
+		if(prob(1))
+			var/datum/effect_system/smoke_spread/bad/smoke = new
+			smoke.set_up(1, loc)
+			smoke.start()
+	if(grilled_item)
+		grill_time += 1
+		grilled_item.reagents.add_reagent(/datum/reagent/consumable/char, 1)
```
Comment (Ghommie) [inline code/modules/food_and_drinks/kitchen_machinery/grill.dm]:
```suggestion
		grilled_item.reagents.add_reagent("char", 1)
```

File: code/modules/food_and_drinks/kitchen_machinery/grill.dm [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+				to_chat(user, "<span class='notice'>There is not enough fuel.</span>")
+				return
+			else if(!grilled_item && user.transferItemToLoc(I, src))
+				grilled_item = I
+				to_chat(user, "<span class='notice'>You put the [grilled_item] on [src].</span>")
+				update_icon()
+				grill_loop.start()
+				return
+		else
+			if(I.reagents.has_reagent(/datum/reagent/consumable/monkey_energy))
```
Comment (Ghommie) [inline code/modules/food_and_drinks/kitchen_machinery/grill.dm]:
```suggestion
			if(I.reagents.has_reagent("monkey_energy"))
```

File: code/modules/food_and_drinks/food/snacks_pastry.dm (hunk 3 above)
Comment (Trilbyspaceclone) [reply code/modules/food_and_drinks/food/snacks_pastry.dm:92]: SHHH

File: code/modules/food_and_drinks/food/snacks_soup.dm (hunk 4 above)
Comment (Trilbyspaceclone) [reply code/modules/food_and_drinks/food/snacks_soup.dm]: I dont plan on having a nothing chem. From what I can tell on TG it does nothing of note other then just being a juice that can be drinked for RP

**Comment by Ghommie** [comment]
Solve conflicts, I'll give it another quick review later for possible last touchs to this PR before it's merged.

**Review by Ghommie — APPROVED** [review]
Re-re-reviewed. Everything seemengly fine except the original bungo slime donut uses the bungo juice which is not implemented yet here. 
Also I'm working on the the reagents id to typepath PR and I do not want to stall other reagents-related PRs because of it.

**Pull request merged**